### DISCUSSION TITLE
Does OpenRadioss contain string expression to function expression evaluation subroutine?

### DISCUSSION THREAD

**Discussion by halfwitbm** [OpenRadioss/OpenRadioss#1747] [Q&A] [answered]
Hi All,

I have two questions about OpenRadioss function in input Dect. I list here:
1. Does OpenRadioss support continuous function definition in input Dect? I only find table format function. 
2. Does OpenRadioss has string expression to function expression evaluation subroutine?  for example, evaluate the string "4*sin(2*pi*x)" to function expression 4*sin(2*pi*x) and accepted by OpenRadioss.

Did somebody have experience related to this? Please share your knowledge with me. Thanks.

Bin

**Comment by MarianBulla** [comment, marked as answer]
Hi Bin,
thanks for your interest in OpenRadioss.
Regarding your questions:
1. Within OpenRadioss, we have only /FUNCT_SMOOTH, which uses an analytic function to describe a smooth behaviour: 
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/74babfcd-940c-4eec-aed0-332d84914571)
Be aware, that the slope of every point is always zero.
The intention of it, is to be used for some ramp-up or -down functions.
Anyway, due to the fact, that OpenRadioss is an explicite dynamic solver, it is not continousless proceeding in time. It has always a discrete timestep, which can be very small, but not zero. Therefore, usually a tabulated function should be sufficient enough. There is no limitation in the number of datapoints within the table formated function. You can define it as smooth as you need it.
2. There is the possibility of using the /PARAMETER/ kyeword for self-written expressions: 
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/6218373f-caa3-4819-8665-2270daaefda4)

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by halfwitbm** [comment]
Thank you very much for the answer. Marian.
